packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,205 @@
+import { Component, h, Prop, State, Watch } from '@stencil/core';
+import { DataPoint, DataStream, MinimalSizeConfig, Primitive, ViewPortConfig } from '../../../utils/dataTypes';
+import { round } from '../../../utils/number';
+import { isNumberDataStream } from '../../../utils/predicates';
+import { getIcons } from '../../charts/common/annotations/iconUtils';
+import { Threshold } from '../../charts/common/types';
+import { NO_VALUE_PRESENT } from '../../common/terms';
+import { DialLoading } from './sc-dial-loading';
+import { sizeContent, StatusProgress, TextSizeConfig } from './util';
+
+const title = (dataStream: { detailedName?: any; name?: any } | null | false) => {
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+  @Prop() valueColor?: string; // css color string
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: MinimalSizeConfig = { width: 0, height: 0 };
+
+  @Prop() isLoading?: boolean = false;
+
+  @State() fontSize: TextSizeConfig = sizeContent.XL;
+
+  getColorByPercent = (percent: number) => {
+    // eslint-disable-next-line no-nested-ternary
+    return percent < 0.3333
+      ? StatusProgress.CRITICAL
+      : percent < 0.6667
+      ? StatusProgress.WARNING
+      : StatusProgress.NORMAL;
+  };
+
+  @Watch('propertyStream')
+  getSize(newVal: DataStream) {
+    const dataLength = newVal.data.length || 0;
+    const textLength =
+      dataLength && (newVal.data[dataLength - 1].y.toString().length || 0) + (newVal.unit?.length || 1);
+
+    if (textLength >= 0 && textLength < 2) {
```

Comment:
is there someway we can have this configuration done as a constant object, rather than a series of if statements?

Reply:
Definition 'LengthToSize'  for a constant object.